cluster/defaults.go

```diff
@@ -713,3 +724,22 @@ func checkVersionNeedsKubeAPIAuditLog(k8sVersion string) (bool, error) {
 	logrus.Debugf("Cluster version [%s] does not need to have kube-api audit log enabled", k8sVersion[1:])
 	return false, nil
 }
+
+func checkVersionNeedsEncryptionDefault(k8sVersion string) (bool, error) {
+	toMatch, err := semver.Make(k8sVersion[1:])
+	if err != nil {
+		return false, fmt.Errorf("Cluster version [%s] can not be parsed as semver", k8sVersion[1:])
+	}
+	logrus.Debugf("Checking if cluster version [%s] needs to have encryption enabled by default", k8sVersion[1:])
+	// encryption turned on by default in k8s 1.18.0 and up
+	clusterDefaultEncryptionRange, err := semver.ParseRange(">=1.18.0-rancher0")
```

Comment:
does this work for version strings of kind "1.18.0-rancher0-1"?

Reply:
Yes, we parse the version string using `server.Make` here:
https://github.com/aiyengar2/rke/blob/baca494cc3ae61074a4355edb73a4e9509ff3a67/cluster/defaults.go#L729

This parses the version string into its Major, Minor, Patch, and PRVersions (i.e. `rancher-0-1`) and does the comparison in that order. Since `"rancher-0-1" >= "rancher-0"`, it works as expected.

Reply:
Thanks for checking!